include/deal.II/meshworker/scratch_data.h

```diff
@@ -0,0 +1,1301 @@
+   * face index and the subface index.
+   *
+   * The user code should be structured without interleaving work on cells and
+   * work on faces.
+   *
+   * Consider, for example, the following snippet of code:
+   *
+   * @code
+   * ScratchData<dim,spacedim> scratch(...);
+   * FEValuesExtractors::Scalar temperature(0);
+   *
+   * for(const auto &cell : dof_handler.active_cell_iterators()){
+   *   const auto &fe_values = scratch.reinit(cell);
+   *   const auto &local_dof_values =
+   *         scratch.extract_local_dof_values("solution", solution);
+   *
+   *   // This will contain all values of the temperature on the cell
+   *   // quadrature points
+   *   const auto &solution_values_cell =
+   *         scratch.get_values("solution", temperature);
+   *
+   *   // Do something with values on the cell
+   *   ...
+   *
+   *   // Now start working on the faces
+   *   for(unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f) {
+   *      const auto &fe_face_values = scratch.reinit(cell, f);
+   *
+   *      // Now we'll have access to the values of the temperature on the faces
+   *      const auto &solution_values_face =
+   *            scratch.get_values("solution", temperature);
+   *
+   *      // Notice how the function call is the same, but the result depends
+   *      // on what was the last reinit() function that was called. In this
+   *      // case, we called reinit(cell, f), triggering internal work on
+   *      // faces.
+   *   }
+   *
+   *   // At this point, we would like to go back and work on cells,
+   *   // for example querying the values of the gradients:
+   *   const auto &solution_gradients =
+   *         scratch.get_gradients("solution", temperature);
+   *
+   *   // This assertion would be triggered in debug mode
+   *   AssertDimension(solution_gradients.size(), quadrature_cell.size());
+   *
+   *   // However, with the above call the content of gradu is the gradient of
+   *   // the temperature on the quadrature points of the last face visited in
+   *   // the previous loop. If you really want to have the values of the
+   *   // gradients on the cell, at this point your only option is to call
+   *   scratch.reinit(cell);
+   *
+   *   // (again) before querying for the gradients:
+   *   const auto &solution_gradients_cell =
+   *         scratch.get_gradients("solution", temperature);
+   *
+   *   // The call to reinit(cell), however, is an expensive one. You
+   *   // should make sure you only call it once per cell by grouping together
+   *   // all queries that concern cells in the same place (right after you
+   *   // call the reinit(cell) method).
+   *   // A similar argument holds for all calls on each face and subface.
+   * }
+   * @endcode
+   *
+   * When using this class, please cite
+   * @code{.bib}
+   * @article{SartoriGiulianiBardelloni-2018-a,
+   * 	Author = {Sartori, Alberto and Giuliani, Nicola and
```

Comment:
Since parts of this publication are finding its way into deal.II proper, can you add this to the part of `publications.html` that talks about publications that describe parts of deal.II? This is now in the `publications-list` repo.